### DISCUSSION TITLE
MS graph access token from FlutterFire signInWithAuthProvider

### DISCUSSION THREAD

**Discussion by fatima03** [firebase/flutterfire#10499] [Q&A] [answered]
I see this issue is closed, but I am not able to understand the solution, as the token I am not able to call MS graph with the access token I am getting.

[Get ms graph access token from flutterfire signInWithAuthProvider method #9505](https://github.com/firebase/flutterfire/issues/9505)

**Comment by prggTheProgrammer** [comment, marked as answer]
Well, the `sigInInWithAuthprovider` method returns a credential like so:
`cred = await FirebaseAuth.instance.signInWithProvider(provider);`
You can than get the oauth access token like so:
`cred.credential!.accessToken!`

You then need to call MS graph with the Authorization header:
```
http.Response res = await http.get(
    Uri.https("graph.microsoft.com", "v1.0/me"),
    headers: {"Authorization": "Bearer $token"});
```
If you still have any questions, feel free to ask.

**Reply by fatima03** [reply]
Thank you so much, it works!
